### PR TITLE
Add timeout to the telemetry sender

### PR DESCRIPTION
**Pull request by troydai**
This is used to avoid uploader from run into the situation when
Applicaiton Insight service is unresponsive.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/6290
This is an **experimental preview** for @microsoft users.

File: src/azure-cli-core/azure/cli/core/telemetry_upload.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def send(self, data_to_send):
         request = HTTPClient.Request(self._service_endpoint_uri, bytearray(request_payload, 'utf-8'),
                                      {'Accept': 'application/json', 'Content-Type': 'application/json; charset=utf-8'})
         try:
-            response = HTTPClient.urlopen(request)
+            response = HTTPClient.urlopen(request, timeout=10)
```
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/telemetry_upload.py]: Is this timeout in seconds?  What happens to the data if we time out?  Is it lost?  Do we hold on to it and try to send it again with the next bundle?

Comment (troydai) [reply src/azure-cli-core/azure/cli/core/telemetry_upload.py]: 1. In seconds
2. Data lost (its telemetry data)
3. We don't retry

YOLO

**Comment by aog2000a** [comment]
Copying from #6265 : I wonder why should the default be to upload telemetry data (what the hell is that for???) when i am just trying to download a blob. This doesn't feel good in "open source" software... guys, why are you uploading OS version, Python version and machine id strings to Azure?? Bad MS habits? I'm pretty upset as you notice...

Please turn that "feature" default off.

**Pull request merged**